quesma/clickhouse/clickhouse.go

```diff
@@ -474,12 +474,34 @@ func (lm *LogManager) generateNewColumns(
 	attrKeys := getAttributesByArrayName(AttributesKeyColumn, attrsMap)
 	attrTypes := getAttributesByArrayName(AttributesValueType, attrsMap)
 	var deleteIndexes []int
+
+	// HACK Alert:
+	// We must avoid altering the table.Cols map and reading at the same time.
+	// This should be protected by a lock or a copy of the table should be used.
+	//
+	newColumns := make(map[string]*Column)
+	for k, v := range table.Cols {
+		newColumns[k] = v
+	}
+
 	for i := range alteredAttributesIndexes {
-		alterTable := fmt.Sprintf("ALTER TABLE \"%s\" ADD COLUMN IF NOT EXISTS \"%s\" Nullable(%s)", table.Name, attrKeys[i], attrTypes[i])
-		table.Cols[attrKeys[i]] = &Column{Name: attrKeys[i], Type: NewBaseType(attrTypes[i]), Modifiers: "Nullable"}
+
+		columnType := ""
+		// Array and Map are not Nullable
+		if strings.Contains(attrTypes[i], "Array") || strings.Contains(attrTypes[i], "Map") {
+			columnType = attrTypes[i]
+		} else {
+			columnType = fmt.Sprintf("Nullable(%s)", attrTypes[i])
+		}
+		alterTable := fmt.Sprintf("ALTER TABLE \"%s\" ADD COLUMN IF NOT EXISTS \"%s\" %s", table.Name, attrKeys[i], columnType)
+
+		newColumns[attrKeys[i]] = &Column{Name: attrKeys[i], Type: NewBaseType(attrTypes[i]), Modifiers: "Nullable"}
```

Comment:
For now, we always adding `Nullable` modifiers which is inconsistent with above logic

Reply:
Right. I  will fix it before a merge.